src/App.js

```diff
@@ -30,36 +30,111 @@ const App = () => {
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
 
+  const [currentPlayer, setCurrentPlayer] = useState(PLAYER_1);
+
   // Wave 2
   // You will need to create a method to change the square
   //   When it is clicked on.
   //   Then pass it into the squares as a callback
+  const onClickCallback = (squareID) => {
+    if (winner){
+      return;
+    }
+
+    let madeMove = false;
+
+    const newSquares = squares.map(row => row.map(position => {
+      if (position.id != squareID){
+        return position;
+      }
+      if (position.value !== ''){
+        return position;
+      }
```

Comment:
Because both conditionals have the same logic when evaluated to true, we can combine them into one using the `||` (or) operator:

```suggestion
      if (position.id != squareID || position.value !== '' ) {
        return position;
      }
 
```